specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+    AppNotificationActivationInfo activationInfo(L"SampleApp", iconUri);
+    AppNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+
+    if (kind == ExtendedActivationKind::Launch)
+    {
+        // App is launched in FG. So intercept toast activators via FG event
+        const auto token = AppNotificationManager::Default().AppNotificationActivated([](const auto&, const AppNotificationActivatedEventArgs& notificationActivatedEventArgs)
+            {
+                ProcessNotificationArgs(notificationActivatedEventArgs);
+            });
+
+        // App does Foreground Stuff Here
+
+        // Cleanup
+        AppNotificationManager::Default().AppNotificationActivated(token);
+    }
+    else if (kind == ExtendedActivationKind::AppNotification)
+    {
+        auto notificationActivatedEventArgs = args.Data().as<AppNotificationActivatedEventArgs>();
+        ProcessNotificationArgs(notificationActivatedEventArgs);
+    }
+
+    return 0;
+}
+```
+
+## Displaying an App Notification
+
+To display a notification, an app needs to define a payload using an xml schema. In addition, the developer can set a tag and group property which allows the app to replace a prior notification with the same tag and group with an updated version of the payload.
+Finally, the ExpiresOnReboot property can also be set to ensure that the notification is automatically deleted from
+the Action Centre after device reboots.
+
+```cpp
+// Display a notification in action centre using a tag and group
+void DisplayNotification(const winrt::Windows::Data::Xml::Dom::XmlDocument& doc)
+{
+    AppNotification notification(doc);
+    // Setting Tag, Group and ExpiresOnReboot properties are optional
+    notification.Tag(L"Tag");
+    notification.Group(L"Group");
+    notification.ExpiresOnReboot(true);
+    AppNotificationManager::Default().Show(notification);
+}
+```
+## Processing an Activation
+
+The app will need to process the activator in response to a User interacting with the notification in Action Centre. 2 common
+scenarios here are
+
+1. Have the app launch in the foreground in a specific UI context OR
+2. Have the app process an action specific behavior (like a button press in the payload body) in the background.
+
+```cpp
+//<toast launch="action=openThread&amp;threadId=92187">
+//  <visual>
+//      <binding template="ToastGeneric">
+//          <text hint-maxLines="1">Jill Bender</text>
+//          <text>Check out where we camped last weekend! Incredible!!</text>
+//          <image placement="hero" src="https://unsplash.it/360/180?image=1043" />
+//      </binding>
+//  </visual>
+//  <actions>
+//      <input id="textBox" type="text" placeHolderContent="reply" / >
+//      <action
+//          content="Send"
+//          imageUri="Assets/Icons/send.png"
+//          hint-inputId="textBox"
+//          activationType="background"
```

Comment:
>   activationType="background" [](http://example.com/codeflow?start=9&length=30)

Important thing to call out: This attribute is ignored for Win32 apps!

Reply:
Added in the description right above the code.